packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeProvider.cs

```diff
@@ -352,5 +1572,35 @@ private CSharpType GetModelArgumentType(CSharpType modelInterface)
 
             return interfaceArgs[0];
         }
+
+        /// <summary>
+        /// Determines if the type requires a null check in serialization.
+        /// </summary>
+        /// <param name="type">The <see cref="CSharpType"/> to validate.</param>
+        /// <returns><c>true</c> if the type requires a null check.</returns>
+        private bool TypeRequiresNullCheckInSerialization(CSharpType type)
+        {
+            if (type.IsCollection)
+            {
+                return true;
+            }
+            else if (type.IsNullable && type.IsValueType) // nullable value type
+            {
+                return true;
+            }
+            else if (!type.IsValueType && type.IsFrameworkType
+                && (type.FrameworkType != typeof(string) || type.FrameworkType != typeof(byte[])))
+            {
+                // reference type, excluding string or byte[]
+                return true;
+            }
+
+            return false;
+        }
+
+        private bool ShouldOverrideMethod()
```

Comment:
should probably be a field since the value doesn't change ever vs calculating multiple times via the method